app/modules/notifications/models.py

```diff
@@ -75,29 +99,53 @@ def __repr__(self):
             ')>'.format(class_name=self.__class__.__name__, self=self)
         )
 
+    @property
+    def owner(self):
+        return self.recipient
+
+    def get_sender_name(self):
+        return self.message_values.get('sender_name', 'N/A')
+
+    def get_sender_email(self):
+        return self.message_values.get('sender_email', 'N/A')
+
     # returns dictionary of channel:bool
-    def channels_to_send(self):
+    def channels_to_send(self, digest=False):
         # pylint: disable=invalid-name
+        # In future the channels to send right now will be different for digest generation
         channels = None
         user_prefs = UserNotificationPreferences.get_user_preferences(self.recipient)
         if self.message_type in user_prefs.keys():
             channels = user_prefs[self.message_type]
 
         return channels
 
+    def send_if_required(self):
+        from app.modules.emails.utils import EmailUtils
+
+        channels = self.channels_to_send(False)
+        if channels[NotificationChannel.email]:
+            outgoing_message = EmailUtils.build_email(
+                self.message_type, self.message_values
+            )
```

Comment:
see my comment above about adding `subject` value to `send_email()` ... i wonder if this should be its own (teeny-tiny plaintext) template that is built out like `outgoing_message`?   so we can have things like "_You have a collaboration request from {user} on {sitename}"_ or whatever.

Reply:
Yes I like the idea, should be easy to do.